File: prm-backend/src/main/java/prm/prmbackend/entity/Creator.java
package prm.prmbackend.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

/**
 * MongoDB collection: creators
 *
 * Indexes managed by MongoIndexConfig (not annotations):
 *   slug — unique
 *   name — text
 *
 * Role (author / artist) is NOT stored here — determined by
 * MangaSeries.authorIds and MangaSeries.artistIds.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "creators")
public class Creator {

    @Id
    private String id;

    private String name;

    /** unique — enforced by MongoIndexConfig */
    private String slug;

    private String originalName;

    private String biography;

    /** URL to avatar image — no binary data stored in MongoDB */
    private String avatarUrl;

    private Instant createdAt;

    private Instant updatedAt;
}

File: lib/data/models/creator.dart
class Creator {
  final String id;
  final String name;
  final String slug;
  final String? originalName;
  final String? biography;
  final String? avatarUrl;

  Creator({
    required this.id,
    required this.name,
    required this.slug,
    this.originalName,
    this.biography,
    this.avatarUrl,
  });

  factory Creator.fromJson(Map<String, dynamic> json) {
    return Creator(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      slug: json['slug'] ?? '',
      originalName: json['originalName'],
      biography: json['biography'],
      avatarUrl: json['avatarUrl'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'slug': slug,
      'originalName': originalName,
      'biography': biography,
      'avatarUrl': avatarUrl,
    };
  }
}

class CreatorRequest {
  final String name;
  final String slug;
  final String? originalName;
  final String? biography;
  final String? avatarUrl;

  CreatorRequest({
    required this.name,
    required this.slug,
    this.originalName,
    this.biography,
    this.avatarUrl,
  });

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'slug': slug,
      'originalName': originalName,
      'biography': biography,
      'avatarUrl': avatarUrl,
    };
  }
}
